src/MatrixTypes_Trilinos.f90

```diff
@@ -196,14 +217,16 @@ SUBROUTINE set_TrilinosMatrixType(matrix,i,j,setval)
       IF(matrix%isInit) THEN
         IF(((j <= matrix%n) .AND. (i <= matrix%n)) &
           .AND. ((j > 0) .AND. (i > 0))) THEN
-          IF(matrix%isAssembled) CALL ForPETRA_MatReset(matrix%A)
+          IF(matrix%isAssembled) CALL matrix%A%resumeFill()
```

Comment:
I think we need this to start changing the matrix again, but reset would deallocate the memory and reallocate it.  The idea with reset is that we want to zero out previous entries.  This might also work.